sway-core/src/semantic_analysis/namespace/root.rs

```diff
@@ -16,6 +16,7 @@ use super::{module::Module, namespace::Namespace, Path};
 #[derive(Clone, Debug)]
 pub struct Root {
     pub(crate) module: Module,
+    pub(crate) pkg_name: Option<Ident>,
```

Comment:
Right now the content of `root.pkg_name` is the same of `root.module.name`, the only difference is that pkg_name is an Ident and module.name is a String. and pkg_name requires a leak such as `root.pkg_name = Some(Ident::new_no_span(Box::leak(pkg_name.into_boxed_str())))` to be created, which should only be one leak per compilation. To avoid more leaks we have this duplicated, but if we deem that a leak per module is acceptable we could remove `pkg_name` and use only `module.name`.